src/services_models/services_constructor.jl

```diff
@@ -106,6 +117,10 @@ function construct_service!(
                 type = typeof(service),
                 name = PSY.get_name(service),
             )].contributing_devices
+        if !isempty(incompatible_device_types)
+            contributing_devices =
+                [d for d in contributing_devices if typeof(d) ∉ incompatible_device_types]
+        end
```

Comment:
Can this code block be removed?  You shouldn't need to pass incompatible_device_types to this function since you already handled it.

Reply:
No, because there might be more devices than the incompatible ones in the contributing devices.

Reply:
Seems like we're missing each other. In the `construct_services` function you already filtered out the incompatible devices.  This check looks redundant.  Same for the other `construct_service` function.

You should probably assert that contributing_devices is not empty, just like you did in the other one.